Source/Core/VideoCommon/AsyncShaderCompiler.cpp

```diff
@@ -65,17 +67,23 @@ bool AsyncShaderCompiler::HasCompletedWork()
   return !m_completed_work.empty();
 }
 
-void AsyncShaderCompiler::WaitUntilCompletion()
+bool AsyncShaderCompiler::WaitUntilCompletion(bool interruptable)
 {
   while (HasPendingWork())
+  {
+    if (interruptable && Core::GetState() == Core::State::Stopping)
+      return false;
+
     std::this_thread::sleep_for(std::chrono::milliseconds(1));
+  }
+  return true;
 }
 
-void AsyncShaderCompiler::WaitUntilCompletion(
-    const std::function<void(size_t, size_t)>& progress_callback)
+bool AsyncShaderCompiler::WaitUntilCompletion(
+    bool interruptable, const std::function<void(size_t, size_t)>& progress_callback)
```

Comment:
interruptable is always going to be true; we can skip the parameter and just always check Core::GetState()